### DISCUSSION TITLE
What is the correct max input for exp()?

### DISCUSSION THREAD

**Discussion by antoniopgs** [PaulRBerg/prb-math#119] [Q&A] [answered]
It looks like the max input for `exp()` is `133.084258667509499441`:

- https://github.com/paulrberg/prb-math/blob/main/contracts/PRBMathUD60x18Typed.sol#L138
- https://github.com/paulrberg/prb-math/blob/main/contracts/PRBMath.sol#L73

However, here it says it's actually `88.722839111672999628`:
- https://github.com/paulrberg/prb-math/blob/main/contracts/PRBMathUD60x18Typed.sol#L130

I'm assuming the former is correct and the latter is wrong. Right?

**Comment by PaulRBerg** [comment, marked as answer]
Good catch! You're right in that the former is correct, i.e. `133.084258667509499441`.

This is a vestige of the original implementation of the `exp` function, which was using 128.128-bit as the internal number representation. Now I'm using 192.64-bit, which allows for better accuracy, hence a greater allowed input. You can read more about the history of this refactor in [issue #32](https://github.com/paulrberg/prb-math/issues/32) and in [commit 6d28ff](https://github.com/paulrberg/prb-math/commit/6d28ff261a9e28583e032a55d33d1b66e48058c3). Such a shame that I didn't create a PR for this.

So this is just a documentation typo. I pushed a commit to fix it:

https://github.com/paulrberg/prb-math/commit/830eb448d188697c833ec26c4f1a4364c3781131

**Comment by PaulRBerg** [comment]
For future reference: PRBMath now explicitly defines the maximum inputs allowed in `exp` and `exp2`:

- [`EXP_MAX_INPUT`](https://github.com/PaulRBerg/prb-math/blob/97170d0f52de76edfb5fe63c5810a8705c3b0ea7/src/ud60x18/Constants.sol#L13)
- [`EXP2_MAX_INPUT`](https://github.com/PaulRBerg/prb-math/blob/97170d0f52de76edfb5fe63c5810a8705c3b0ea7/src/ud60x18/Constants.sol#L17)
